firefox-ios/Client/Frontend/Browser/SearchEngines/SearchEngines.swift

```diff
@@ -43,10 +43,19 @@ class SearchEngines {
 
     init(prefs: Prefs, files: FileAccessor, engineProvider: SearchEngineProvider = DefaultSearchEngineProvider()) {
         self.prefs = prefs
-        // By default, show search suggestions
+
         self.shouldShowSearchSuggestions = prefs.boolForKey(
             PrefsKeys.SearchSettings.showSearchSuggestions
         ) ?? true
+        shouldShowBrowsingHistorySuggestions = prefs.boolForKey(
+            PrefsKeys.SearchSettings.showFirefoxBrowsingHistorySuggestions
+        ) ?? true
+        shouldShowBookmarksSuggestions = prefs.boolForKey(
+            PrefsKeys.SearchSettings.showFirefoxBookmarksSuggestions
+        ) ?? true
+        shouldShowSyncedTabsSuggestions = prefs.boolForKey(
+            PrefsKeys.SearchSettings.showFirefoxSyncedTabsSuggestions
+        ) ?? true
```

Comment:
nit: As we keep adding stuff here, this function is growing and growing. This whole block of code seems like it could live in a private function called `initPrefBasedSuggestions` or something, that the `init` just calls. Makes it easier to read and keeps things small and compact.